AutoReplayUploader/AutoReplayUploaderPlugin.cpp

```diff
@@ -5,194 +5,341 @@
-			if (*showNotifications) gameWrapper->Toast("Autoreplayuploader", "Cannot upload to ballchasing.com, no authkey set!", "ballchasing_logo", 3.5f, ToastType_Error);
-		}
-		else 
-		{
-			std::string visibility = cvarManager->getCvar("cl_autoreplayupload_ballchasing_visibility").getStringValue();
-			UploadReplayToEndpoint(replayPath, GenerateUrl(BALLCHASING_ENDPOINT_DEFAULT, { {"visibility", visibility} }), "file", authKey, "ballchasing.com");
-		}
+		UploadReplayToEndpoint(replayPath, BALLCHASING);
 	}
 	CheckFileUploadProgress(gameWrapper.get());
 }
 
+/**
+* SetReplayName - Called to set the name of the replay in the replay file.
+* Params:
+*	server - ServerWrapper
+*	soccarReplay - Replay to set name of
+*	replayName - A templatized string that accepts the following tokens for replacement.
+*		{PLAYER} - Name of current steam user
+*		{MODE} - Game mode of replay (Private, Ranked Standard, etc...)
+*		{NUM} - Current sequence number to allow for uniqueness
+*		{YEAR} - Year since 1900 % 100, eg. 2019 returns 19
+*		{MONTH} - Month 1-12
+*		{DAY} - Day of the month 1-31
+*		{HOUR} - Hour of the day 0-23
+*		{MIN} - Min of the hour 0-59
+*		{WINLOSS} - W or L depending on if the player won or lost
+*/
+void AutoReplayUploaderPlugin::SetReplayName(ServerWrapper& server, ReplaySoccarWrapper& soccarReplay, std::string replayName)
+{
+	// Get Gamemode game was in
+	auto playlist = server.GetPlaylist();
+	auto mode = GetPlaylistName(playlist.GetPlaylistId());
+
+	// Get current Sequence number
+	auto seq = std::to_string(*templateSequence);
+	*templateSequence = *templateSequence + 1; // increment to next sequence number
+
+	// Get date string
+	auto t = std::time(0);
+	auto now = std::localtime(&t);
+
+	// Calculate Win/Loss string
+	auto team = server.GetGameWinner();
+	auto won = team.GetTeamIndex() == 0 ? soccarReplay.GetTeam0Score() > soccarReplay.GetTeam1Score() : soccarReplay.GetTeam1Score() > soccarReplay.GetTeam0Score();
+	auto winloss = won ? string("W") : string("L");
+
+	cvarManager->log("Username: " + steamUserName);
+	cvarManager->log("Mode: " + mode);
+	cvarManager->log("Sequence: " + seq);
+	cvarManager->log("Date: " + std::to_string(now->tm_year % 100) + "-" + std::to_string(now->tm_mday) + "-" + std::to_string(now->tm_year % 100) + " " + std::to_string(now->tm_hour) + ":" + std::to_string(now->tm_min));
+	cvarManager->log("WinLoss: " + winloss);
+
+	ReplaceAll(replayName, "{PLAYER}", steamUserName);
+	ReplaceAll(replayName, "{MODE}", mode);
+	ReplaceAll(replayName, "{NUM}", seq);
+	ReplaceAll(replayName, "{YEAR}", std::to_string(now->tm_year % 100));
+	ReplaceAll(replayName, "{MONTH}", std::to_string(now->tm_mon + 1));
+	ReplaceAll(replayName, "{DAY}", std::to_string(now->tm_mday));
+	ReplaceAll(replayName, "{HOUR}", std::to_string(now->tm_hour));
+	ReplaceAll(replayName, "{MIN}", std::to_string(now->tm_min));
+	ReplaceAll(replayName, "{WINLOSS}", winloss);
```

Comment:
Any suggestions for other replacements?

Reply:
Create a separate `{WINLOSS} ` and `{WL} `. WL shows just the letters W or L whilst winloss shows "Win" or "Loss"?

Reply:
Maybe {MMR} (should be retrievable from MMR wrapper (although this might not work if user doesn't have show MMR enabled (needs testing if this is considered))

Reply:
Replacement suggestions:
* Teammate names
* Opposing team member names

:)

Reply:
MMR, Teammate names, Opposing team member names could all be added although I think the member names could get long but perhaps that is the user's choice.

But they would all need additional testing cause I've found that calling functions inside of serverwrappers often crash rocket league... lol  Since I don't have access to actually run rocket league at the moment I will take these as considerations to add in a future PR.